pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java

```diff
@@ -680,10 +686,175 @@ public Reader getCharacterStream(String parameterName) throws SQLException {
+      } else {
+        throw new PSQLException(GT.tr("conversion to {0} from {1} not supported", type, sqlType),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Timestamp.class) {
+      if (sqlType == Types.TIMESTAMP
+              //#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.2"
+              || sqlType == Types.TIMESTAMP_WITH_TIMEZONE
+      //#endif
+      ) {
+        return type.cast(getTimestamp(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("conversion to {0} from {1} not supported", type, sqlType),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Calendar.class) {
+      if (sqlType == Types.TIMESTAMP) {
+        Timestamp timestampValue = getTimestamp(parameterIndex);
+        Calendar calendar = Calendar.getInstance(getDefaultCalendar().getTimeZone());
+        calendar.setTimeInMillis(timestampValue.getTime());
+        return type.cast(calendar);
+      } else {
+        throw new PSQLException(GT.tr("conversion to {0} from {1} not supported", type, sqlType),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == java.util.Date.class) {
+      if (sqlType == Types.TIMESTAMP) {
+        Timestamp timestamp = getTimestamp(parameterIndex);
+        return type.cast(new java.util.Date(timestamp.getTime()));
+      } else {
+        throw new PSQLException(GT.tr("conversion to {0} from {1} not supported", type, sqlType),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Array.class) {
+      if (sqlType == Types.ARRAY) {
+        return type.cast(getArray(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("conversion to {0} from {1} not supported", type, sqlType),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == SQLXML.class) {
+      if (sqlType == Types.SQLXML) {
+        return type.cast(getSQLXML(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("conversion to {0} from {1} not supported", type, sqlType),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == UUID.class) {
+      return type.cast(getObject(parameterIndex));
+    } else if (type == InetAddress.class) {
+      Object addressString = getObject(parameterIndex);
+      if (addressString == null) {
+        return null;
+      }
+      try {
+        return type.cast(InetAddress.getByName(((PGobject) addressString).getValue()));
+      } catch (UnknownHostException e) {
+        throw new SQLException("could not create inet address from string '" + addressString + "'");
+      }
+    } else if (type == ResultSet.class) {
       return type.cast(getObject(parameterIndex));
     }
-    throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+    throw new PSQLException(GT.tr("conversion to {0} from {1} not supported", type, sqlType),
```

Comment:
This likely breaks existing localized strings (if any)

Reply:
So, should I revert the exception message so that is stays compatible or should I try my best to update the localizations?

Reply:
OK, let me try to find the translators for the main project to see

Reply:
```
pgjdbc/src/main/java/org/postgresql/translation//ru.po:msgid "Unsupported type conversion to {1}."
pgjdbc/src/main/java/org/postgresql/translation//ru.po-msgstr "Бинарная передача не поддерживается для типа  {0}"
```
Russian text here says "binary transfer is not supported for {0}"

```
pgjdbc/src/main/java/org/postgresql/translation//ru.po:msgid "conversion to {0} from {1} not supported"
pgjdbc/src/main/java/org/postgresql/translation//ru.po-msgstr "Уровень изоляции транзакций {0} не поддерживается."
```
Russian text says "transaction isolation level {0} is not supported"

The translations are completely off.

It looks like something wrong went with translation messages :-(

Reply:
It looks like messages were corrupted in https://github.com/pgjdbc/pgjdbc/commit/e5aab1cd3e49051f46298d8f1fd9f66af1731299#diff-c570cb92bee10b33b00278918b2888b2R1518  (REL42.2.2+)